### PR TITLE
fix: [SRM-10674]: DSL upgrade and Stackdriver test case

### PR DESCRIPTION
**Pull request by AbhijithHarness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34952)

### CODE REVIEW

File: project/main_maven_install.json [hunk 1]
```diff
@@ -11153,14 +11153,14 @@
                     "com.squareup.retrofit2:converter-gson:2.3.0",
                     "com.google.guava:guava:31.1-jre"
                 ],
-                "file": "v1/https/harness.jfrog.io/artifactory/harness-internal/io/harness/cv/data-collection-dsl/0.37-RELEASE/data-collection-dsl-0.37-RELEASE.jar",
+                "file": "v1/https/harness.jfrog.io/artifactory/harness-internal/io/harness/cv/data-collection-dsl/0.38-RELEASE/data-collection-dsl-0.38-RELEASE.jar",
```
[the remark this answers is not in the document]
Comment (AbhijithHarness) [reply project/main_maven_install.json:11156]: Yes, added null check in DSL code.

**Review by kapil-harness — COMMENTED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34952)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by kapil-harness — APPROVED** [review]

**Review by kapil-harness — COMMENTED** [review]
approve

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34952#-:-N67ZI8YCL9QjzqrioYT:bmf6je0)* status: 1. unresolved discussions.

**Review by kapil-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34952#-:-N67ZNXt4_lnSqTj0zY6:bnfp4nl)* status: 1. unresolved discussions.

**Review by kapil-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34952)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[`project/main_maven_install.json` line 11156 at r1](https://reviewable.io/reviews/harness/harness-core/34952#-N66qmurYZ4SJHGqi6k4-r1-11156:-N67ZVF_7M6Ri8iKcewU:b-f8humc) ([raw file](https://github.com/harness/harness-core/blob/29a5415d32d3276424f76245f473d18159705b1f/project/main_maven_install.json#L11156)):*
<details><summary><i>Previously, AbhijithHarness (Abhijith Pradeep) wrote…</i></summary><blockquote>

Yes, added null check in DSL code.
</blockquote></details>

approved

**Review by kapil-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34952)* status: :shipit: complete! ✓

___
*[`project/main_maven_install.json` line 11156 at r1](https://reviewable.io/reviews/harness/harness-core/34952#-N66qmurYZ4SJHGqi6k4-r1-11156:-N67ZffM1a94n9_YOnxP:b-hcfagp) ([raw file](https://github.com/harness/harness-core/blob/29a5415d32d3276424f76245f473d18159705b1f/project/main_maven_install.json#L11156)):*
<details><summary><i>Previously, AbhijithHarness (Abhijith Pradeep) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

resolve

**Review by kapil-harness — APPROVED** [review]

**Pull request merged**